src/MinterGateway.sol

```diff
@@ -1066,58 +1074,88 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
         bytes[] calldata signatures_
-    ) internal view returns (uint40 minTimestamp_) {
-        uint256 threshold_ = updateCollateralValidatorThreshold();
-
+    ) internal returns (uint40 minTimestamp_) {
         minTimestamp_ = uint40(block.timestamp);
 
-        // Stop processing if there are no more signatures or `threshold_` is reached.
-        for (uint256 index_; index_ < signatures_.length && threshold_ > 0; ++index_) {
+        uint256 validCount_;
+
+        for (uint256 index_; index_ < signatures_.length; ++index_) {
             unchecked {
                 // Check that validator address is unique and not accounted for
                 // NOTE: We revert here because this failure is entirely within the minter's control.
                 if (index_ > 0 && validators_[index_] <= validators_[index_ - 1]) revert InvalidSignatureOrder();
             }
 
-            // Check that validator is approved by TTG.
-            if (!isValidatorApprovedByTTG(validators_[index_])) continue;
-
-            // Check that the timestamp is not 0.
-            if (timestamps_[index_] == 0) revert ZeroTimestamp();
-
-            // Check that the timestamp is not in the future.
-            if (timestamps_[index_] > uint40(block.timestamp)) revert FutureTimestamp();
-
-            // NOTE: Need to store the variable here to avoid a stack too deep error.
-            bytes32 digest_ = _getUpdateCollateralDigest(
-                minter_,
-                collateral_,
-                retrievalIds_,
-                metadataHash_,
-                timestamps_[index_]
-            );
-
-            // Check that ECDSA or ERC1271 signatures for given digest are valid.
-            if (!SignatureChecker.isValidSignature(validators_[index_], digest_, signatures_[index_])) continue;
+            if (
+                !_verifyValidatorSignature(
+                    minter_,
+                    collateral_,
+                    retrievalIds_,
+                    metadataHash_,
+                    validators_[index_],
+                    timestamps_[index_],
+                    signatures_[index_]
+                )
+            ) continue;
 
             // Find minimum between all valid timestamps for valid signatures.
             minTimestamp_ = UIntMath.min40(minTimestamp_, uint40(timestamps_[index_]));
 
             unchecked {
-                --threshold_;
+                ++validCount_;
             }
         }
 
-        // NOTE: Due to STACK_TOO_DEEP issues, we need to refetch `requiredThreshold_` and compute the number of valid
-        //       signatures here, in order to emit the correct error message. However, the code will only reach this
-        //       point to inevitably revert, so the gas cost is not much of a concern.
-        if (threshold_ > 0) {
-            uint256 requiredThreshold_ = updateCollateralValidatorThreshold();
+        uint256 requiredThreshold_ = updateCollateralValidatorThreshold();
```

Comment:
Can be inlined since in most cases we won't revert.

Reply:
Gets multi-lined longer then, and I don't think it optimizes.